draft-thomson-use-it-or-lose-it.md

```diff
@@ -379,13 +377,32 @@ the mechanisms it safeguards.  Though it has been effective at revealing
 problems in some cases with TLS, its efficacy isn't proven more generally.
 
 This style of defensive design is limited because it is only superficial.  It
-only exercises a small part of the mechanisms supporting extensibility.  More
+only exercises a small part of the mechanisms that support extensibility.  More
 critically, it does not easily translate to all forms of extension point.  For
 instance, HMSV negotiation cannot be greased in this fashion.  Other techniques
 might be necessary for protocols that don't rely on the particular style of
 exchange that is predominant in TLS.
 
 
+## Invariants
+
+Documenting aspects of the protocol that cannot or will not change as
+extensions or new versions are added can be a useful exercise. Understanding
+what aspects of a protocol are invariant can help guide the process of
+protecting those parts of the protocol that might change.
+
+Like greasing, a declaration of protocol invariants is useful only to the
+extent that implementations of the protocol are willing to admit the
+possibility of new uses for the protocol.  A protocol that declares protocol
+invariants relies on implementations understanding and respecting those
+invariants.
```

Comment:
Not sure I really understand what you want to say here. When you talk about implementations, is that endpoint implementations or any implementation that may interact with the protocol? Also how could an implementation not admit an invariant? For me documenting invariants was not really intended to guide implementation (so far) but rather have an explicit step in the design of a protocol.

Reply:
I took another spin at this.  Let me know if this is better.